packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -585,11 +607,15 @@ Hardhat Network's forking functionality only works with blocks from at least spu
   }
 
   public async getLatestBlock(): Promise<Block> {
+    if (this._blockchain.isReservedBlock("latest")) {
```

Comment:
The reason I wrote it this way is because the `Common` needs to be passed into the creation of the block, via ethereumjs' `Block.fromBlockData()`, and the node is the only level of abstraction that already has a common handy.

I can't remember exactly *why* we need to pass the common into `fromBlockData()`, but I have a vague memory of it relating to having the proper hardfork set on the block, and if you comment out that argument from the `fromBlockData()` call, there will be some test failures with errors like `calcNextBaseFee() can only be called with EIP1559 being activated`.

The only way I can see to take your suggestion would be to store a copy of the common within the `HardhatBlockchain` class, but that seems counter-intuitive to me, but I'm open to suggestions!

Reply:
I talked to Pato about this. Keeping the Common as member data within the Blockchain classes feels weirder but also safer and less error prone, so we're going to opt for that approach.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/2032/commits/51e2d3eccb262e2e56a6d6842f610dd606a17c15